lib/test_prof/factory_default.rb

```diff
@@ -32,27 +32,28 @@ def init
         TestProf::FactoryBot::Strategy::Stub.prepend StrategyExt
 
         @store = {}
-        @store_preserve_traits = {}
         # default is false to retain backward compatibility
         @preserve_traits = false
       end
 
-      def register(name, obj, preserve_traits = nil)
-        unless FactoryDefault.preserve_traits
-          @store_preserve_traits[name] ||= true if preserve_traits
-        end
-        store[name] = obj
+      def register(name, obj, **options)
+        options[:preserve_traits] = true if FactoryDefault.preserve_traits
+        store[name] = { object: obj, **options }
+        obj
       end
 
-      def get(name, _traits = nil)
-        store[name]
-      end
+      def get(name, traits = nil)
+        record = store[name]
+        return unless record
 
-      def exists?(name, traits = nil)
         if traits && !traits.empty?
-          return false if FactoryDefault.preserve_traits || @store_preserve_traits[name]
+          return nil if FactoryDefault.preserve_traits || record[:preserve_traits]
         end
-        store.key?(name)
+        record[:object]
+      end
+
+      def exists?(name, traits = nil)
```

Comment:
Do we use `exists?` anywhere?

Reply:
No, we don't, can be safely removed - left it for compatibility. It was public and appeared in generated documentation, someone could have used it for some test suite debugging. (sure I doubt it too)

Reply:
Let's remove then

Reply:
Ok, done